homeassistant/components/sensor/__init__.py

```diff
@@ -536,39 +605,49 @@ def state(self) -> Any:
                     report_issue,
                 )
                 return value
+        else:
+            numerical_value = value
+
+        if (
+            prec is not None
+            and (self._sensor_option_precision or not isinstance(value, int))
+            and (
+                native_unit_of_measurement == unit_of_measurement
+                or device_class not in UNIT_CONVERTERS
+            )
+        ):
+            value = f"{numerical_value:.{prec}f}"
 
         if (
             native_unit_of_measurement != unit_of_measurement
             and device_class in UNIT_CONVERTERS
         ):
+            prec = self.precision
```

Comment:
No need to set variable again (already set on line 568)
I suggest to replace with a comment
```suggestion
            # Unit conversion needed
```